### PR TITLE
Fix duplicate channels added by channel pins

### PR DESCRIPTION
**Pull request by tl-hbk**
### Description

While debugging https://github.com/conda/conda-libmamba-solver/issues/447 I noticed that channel pins would add duplicate channels.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda_libmamba_solver/solver.py [hunk 1]
```diff
@@ -198,7 +198,11 @@ def solve_final_state(
                     all_channels.append(channel)
         all_channels.extend(in_state.maybe_free_channel())
 
-        all_channels = tuple(all_channels)
+        distinct_channels = OrderedDict()
+        for c in all_channels:
+            if c not in distinct_channels:
+                distinct_channels[c] = True
+        all_channels = tuple(distinct_channels.keys())
```
Comment (jaimergp) [inline conda_libmamba_solver/solver.py:205]: You can dedup while keeping order with:

```suggestion
        all_channels = tuple(dict.fromkeys(all_channels))
```

Comment (tl-hbk) [reply conda_libmamba_solver/solver.py]: Thanks for the suggestion I didn't know that.  I've gone ahead and changed it to this

**Comment by jaimergp** [comment]
CI is broken (I need to fix a couple things after some changes upstream), but this looks sensible to me. Thanks!

**Review by jaimergp — APPROVED** [review]

**Comment by jaimergp** [comment]
Thanks @tl-hbk!

**Pull request merged**